yarn-project/archiver/src/archiver/kv_archiver_store/log_store.ts

```diff
@@ -223,19 +230,20 @@ export class LogStore {
    * that tag.
    */
   getLogsByTags(tags: Fr[]): Promise<TxScopedL2Log[][]> {
-    return this.db.transaction(() =>
-      tags
-        .map(tag => this.#logsByTag.get(tag.toString()))
-        .map(noteLogBuffers => noteLogBuffers?.map(noteLogBuffer => TxScopedL2Log.fromBuffer(noteLogBuffer)) ?? []),
-    );
+    return this.db.transactionAsync(async () => {
+      const logs = await Promise.all(tags.map(tag => this.#logsByTag.getAsync(tag.toString())));
+      return logs.map(
+        noteLogBuffers => noteLogBuffers?.map(noteLogBuffer => TxScopedL2Log.fromBuffer(noteLogBuffer)) ?? [],
+      );
+    });
```

Comment:
Shouldn't this one be a read tx?

Reply:
Yes, thanks. We have been trying to find all the places where we have unnecessary write transactions. We missed this one.